### PR TITLE
Share code online With Oscript Editor.

### PR DESCRIPTION
**Pull request by genievot**
### **Added a sharing code button to make sharing online easy for users.**

Users can share the code from Oscript editor to directly show it's working to anyone. The code-sharing is powered with free API http://myjson.com which will take the whole agent object and create a router URL for it which can be copied and given to anybody. This will open the website with prewritten AA in it.

Because of History mode in Vue, I am using one of the recommended [Firebase hosting](https://cli.vuejs.org/guide/deployment.html#firebase) to tackle the router problem...
**The demo is here to test out:-** https://oscript-editor.firebaseapp.com/

I would like to request that I am not that good with writing the code with best practices, so please adjust it accordingly if you want to or if you think that there is a need for it.
Thank you.

### CODE REVIEW

**Comment by tonyofbyteball** [comment]
This is a nice feature!  Could you additionally copy the URL to clipboard and briefly display a message like "Copied link to clipboard".  Currently, there is no visual feedback after one clicks the share button and it's not quite evident that the URL changed.  Otherwise, great work!

**Comment by genievot** [comment]
> This is a nice feature! Could you additionally copy the URL to clipboard and briefly display a message like "Copied link to clipboard". Currently, there is no visual feedback after one clicks the share button and it's not quite evident that the URL changed. Otherwise, great work!

Sure great thought, I will do it as soon as possible and add a new commit. Thank you.

**Comment by genievot** [comment]
Done @tonyofbyteball  please take a look and let me know if something to change. I used simple alert instead of any new component to showing a notification to the user.

**Comment by genievot** [comment]
Fixed the uncommon behavior of adding duplicate labels for the same text.

**Comment by genievot** [comment]
Added serial number incrementation in this commit which means if code label is already there and someone opens a shared link with same label name then it will be like this...




and no wired id will be showing to make it unique.

File: src/store/modules/lib/agents.js [hunk 1]
```diff
@@ -57,11 +60,91 @@ export default () => ({
...
+					text
+				})
+			} else {
+				let findOldAgentText = state.userAgents.find(a => a.text === text)
+				let findOldAgentLabel = state.userAgents.find(a => a.label === label)
+				if (findOldAgentText) {
+					findOldAgentText.text = text
+					findOldAgentText.id = id
+					// This below condition checks. If there is already named label and the old text and if that text === label's state text
+				} else if (findOldAgentLabel && findOldAgentText && findOldAgentLabel.text === findOldAgentText.text) {
```
Comment (tonyofbyteball) [inline src/store/modules/lib/agents.js]: this in never true after `if (findOldAgentText)` above

File: src/store/modules/lib/agents.js [hunk 2]
```diff
@@ -57,11 +60,91 @@ export default () => ({
...
+					findOldAgentLabel.text = text
+					findOldAgentLabel.id = id
+				} else if (findOldAgentLabel) {
+					var inBrackets = findOldAgentLabel.label.trim().match(/\(.*?\)/g)
+					if (findOldAgentLabel || inBrackets) {
```
Comment (tonyofbyteball) [inline src/store/modules/lib/agents.js]: this is always true due to the condition 2 lines above

File: src/store/modules/lib/agents.js [hunk 3]
```diff
@@ -57,11 +60,91 @@ export default () => ({
...
+						if (inBrackets) {
+							var wasLabel = findOldAgentLabel.label
```
Comment (tonyofbyteball) [inline src/store/modules/lib/agents.js]: consider using the existing function getIncrementedLabel instead

File: index.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+const electron = require('electron')
```
Comment (tonyofbyteball) [inline index.js]: please keep this PR focused on one subject, move electron to a separate PR

Comment (genievot) [reply index.js]: oops didn't mean to add it, Don't know how it ended up there. Will remove it and fix remaining reviews.

File: src/store/modules/lib/agents.js (hunk 1 above)
Comment (genievot) [reply src/store/modules/lib/agents.js]: yes, good catch should have followed this.

File: src/store/modules/lib/agents.js (hunk 3 above)
Comment (genievot) [reply src/store/modules/lib/agents.js]: The existing function cannot be used right now. But in future with time, I will try to manage it.

**Comment by genievot** [comment]
Fixed the code problems and removed index.js for electron. But the use of current function `getIncrementedLabel` Cannot be used right now. The method checks for integer inside brackets but checks for postfix integer.
